docs/UpdatingMajorVersionAndTFM.md

```diff
@@ -0,0 +1,37 @@
+# Updating to a new Major Version & TFM
+
+At the end of each release cycle, we update our Major Version & TFM (Target Framework Moniker) in `main`, in preparation for the next major release. This doc describes the process of doing those updates, which can include many subtle gotchas in the aspnetcore repo.
+
+## Updating Major Version
+
+Typically, we will update the Major Version before updating the TFM. This is because updating Major Version only requires a logical agreement that `main` is no longer the place for work on the previous major release, while updating the TFM requires waiting for dotnet/runtime to update their TFM so that we can ingest that change. For an example, [this](https://github.com/dotnet/aspnetcore/pull/35402) is the PR where we updated our branding from 6.0.0 to 7.0.0.
+
+### Required changes
+
+* In [eng/Versions.props](/eng/Versions.props):
+  1. Increment `AspNetCoreMajorVersion` by 1.
+  2. Change `PreReleaseVersionIteration` to `1`.
+  3. Change `PreReleaseVersionLabel` to `alpha`.
+  4. Change `PreReleaseBrandingLabel` to `Alpha $(PreReleaseVersionIteration)`.
+* In [src/Framework/test/TestData.cs](/src/Framework/test/TestData.cs), update `ListedTargetingPackAssemblies` by incrementing the AssemblyVersion of all aspnetcore assemblies by 1 major version. Once dotnet/runtime updates their AssemblyVersions, we also need to update those in this file. They typically make that change at the same time as their TFM update, but we change our AssemblyVersions as soon as we update branding.
+* Add entries to [nuget.config](/nuget.config) for the new Major Version's feed. This just means copying the current feeds (e.g. `dotnet7` and `dotnet7-transport`) and adding entries for the new feeds (`dotnet8` and `dotnet8-transport`). Make an effort to remove old feeds here at the same time.
+* In [src/ProjectTemplates/Shared/TemplatePackageInstaller.cs](/src/ProjectTemplates/Shared/TemplatePackageInstaller.cs), add entries to `_templatePackages ` for `Microsoft.DotNet.Web.ProjectTemplates` and `Microsoft.DotNet.Web.Spa.ProjectTemplates` matching the new version.
```

Comment:
Is there a validation that you can run after this point to make sure you did everything right? I assume a full test run is sufficient enough validation?

Reply:
Good point on validation - full test run is generally sufficient, though you should also check that the build produces packages with the new version. I'll add that.